xgcm/grid.py

```diff
@@ -868,6 +868,20 @@ def transform(
                 "`transform` can only be used on axes that are non-periodic. Pass `periodic=False` to `xgcm.Grid`."
             )
 
+        # complain if the target values are not provided as xr.dataarray
+        for var_name, variable, allowed_types in [
+            ("da", da, [xr.DataArray]),
+            ("target", target, [xr.DataArray, np.ndarray]),
+            ("target_data", target_data, [xr.DataArray]),
+        ]:
+            if not (
+                any([isinstance(variable, ty) for ty in allowed_types])
```

Comment:
`isinstance` allows checking against a tuple of types. So, the second for loop could be removed: 


```suggestion
            ("da", da, (xr.DataArray,)),
            ("target", target, (xr.DataArray, np.ndarray)),
            ("target_data", target_data, (xr.DataArray,)),
        ]:
            if not isinstance(variable, allowed_types)
```

Reply:
oh sorry, didnt see that you already put in a suggestion. I did implement a similar fix below.

Reply:
No worries. The fix looks 👍🏽